### PR TITLE
Validate proxy requests in absolute-form

### PR DESCRIPTION
**Pull request by clue**
The `absolute-form` `request-target` was already supported in earlier version, but was neither tested nor documented. This PR makes sure that `getUri()` behaves consistent across different forms of request-target and adds matching documentation and an example for a plain HTTP proxy that relies on requests in `absolute-form`.

Example request as sent to a plain HTTP proxy:

```
GET http://www.google.com/ HTTP/1.1
Host: www.google.com
```

Builds on top of #167, #158 and #157.

### CODE REVIEW

**Comment by clue** [comment]
Updated PR to remove unrelated changes so this now has a clear focus on requests in `absolute-form` :shipit:

File: tests/ServerTest.php [hunk 1]
```diff
@@ -267,6 +267,121 @@ public function testRequestNonConnectWithAuthorityRequestTargetWillReject()
...
+
+        $data = "GET http://example.com/test HTTP/1.1\r\nHost: example.com\r\n\r\n";
+        $this->connection->emit('data', array($data));
+
+        $this->assertInstanceOf('RingCentral\Psr7\Request', $requestAssertion);
+        $this->assertSame('GET', $requestAssertion->getMethod());
+        $this->assertSame('http://example.com/test', $requestAssertion->getRequestTarget());
+        $this->assertEquals('http://example.com/test', $requestAssertion->getUri());
+        $this->assertSame('/test', $requestAssertion->getUri()->getPath());
+        //$this->assertSame(array(), $requestAssertion->getQueryParams());
```
Comment (WyriHaximus) [inline tests/ServerTest.php]: Any reason this is commented out? To me commented out code shouldn't be committed tbh

File: tests/ServerTest.php [hunk 2]
```diff
@@ -267,6 +267,121 @@ public function testRequestNonConnectWithAuthorityRequestTargetWillReject()
...
+
+        $data = "GET http://example.com:8080/test HTTP/1.0\r\n\r\n";
+        $this->connection->emit('data', array($data));
+
+        $this->assertInstanceOf('RingCentral\Psr7\Request', $requestAssertion);
+        $this->assertSame('GET', $requestAssertion->getMethod());
+        $this->assertSame('http://example.com:8080/test', $requestAssertion->getRequestTarget());
+        $this->assertEquals('http://example.com:8080/test', $requestAssertion->getUri());
+        $this->assertSame('/test', $requestAssertion->getUri()->getPath());
+        //$this->assertSame(array(), $requestAssertion->getQueryParams());
```
Comment (WyriHaximus) [inline tests/ServerTest.php]: Same as line 289

**Comment by clue** [comment]
Thanks for spotting! Updated and also rebased now that #170 is in, which did not preserve original request-target :shipit:

**Comment by WyriHaximus** [comment]
:+1:

**Pull request merged**